README.md

```diff
@@ -23,7 +37,9 @@ appropriate test framework support into your tests.
 `test_*` function in the subclasses.
 
 ### pytest
-`import appmap.pytest`. Instruments and records `test_*` functions.
+`import appmap.pytest`. Instruments and records `test_*` functions. Resulting AppMaps are
```

Comment:
It didn't work for me with `import appmap.pytest` in the test file.

I was able to get it to work by placing `pytest_plugins = ['appmap.pytest']` in the conftest.py

Will investigate if I get some time

Reply:
https://docs.pytest.org/en/stable/writing_plugins.html#conftest-py-local-per-directory-plugins

pytest loads plugin modules at tool startup in the following way:

1. by scanning the command line for the -p no:name option and blocking that plugin from being loaded (even builtin plugins can be blocked this way). This happens before normal command-line parsing.
2. by loading all builtin plugins.
3. by scanning the command line for the -p name option and loading the specified plugin. This happens before normal command-line parsing.
4. by loading all plugins registered through setuptools entry points.
5. by loading all plugins specified through the PYTEST_PLUGINS environment variable.
6. by loading all conftest.py files as inferred by the command line invocation:
    - if no test paths are specified, use the current dir as a test path
    - if exists, load conftest.py and test*/conftest.py relative to the directory part of the first test path. After the conftest.py file is loaded, load all plugins specified in its pytest_plugins variable if present.
  
    Note that pytest does not find conftest.py files in deeper nested sub directories at tool startup. It is usually a good idea to keep your conftest.py file in the top level test or project root directory.

7. by recursively loading all plugins specified by the pytest_plugins variable in conftest.py files.

Reply:
Three options to include appmap in pytest I can think of:

1. Place `pytest_plugins = ['appmap.pytest']` in _conftest.py_
2. Run with `APPMAP=true ./.venv/bin/pytest -p appmap.pytest -s`
3. `PYTEST_PLUGINS=appmap.pytest APPMAP=true ./.venv/bin/pytest -s`

Reply:
pytest-django might be something to look at

Reply:
Yeah, the pytest integration is done by providing a pytest plugin. I'll fix the doc.

Once it's available through PyPI, you'll be able to `pip install appmap` into a project that uses pytest and the appmap plugin will get used.

To test it locally, in a shell that's had a virtual environment activated, you can run `poetry install` in the root of your `appmap-python` repo:

```
ajp@Alans-MacBook-Pro appmap-python % pyenv activate
pyenv-virtualenv: version `flaskbb' is already activated
(flaskbb) params_20210202 *
ajp@Alans-MacBook-Pro appmap-python % pwd
/Users/ajp/src/applandinc/appmap-python
(flaskbb) params_20210202 *
ajp@Alans-MacBook-Pro appmap-python % poetry install
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: appmap (0.0.0)
(flaskbb) params_20210202 *
ajp@Alans-MacBook-Pro appmap-python % pip show appmap
Name: appmap
Version: 0.0.0
Summary: Create AppMap files by recording a Python application.
Home-page: None
Author: Alan Potter
Author-email: alan@app.land
License: None
Location: /Users/ajp/.pyenv/versions/3.9.0/envs/flaskbb/lib/python3.9/site-packages
Requires: orjson, inflection, PyYAML
Required-by:
(flaskbb) params_20210202 *
ajp@Alans-MacBook-Pro appmap-python % pytest --collect-only | head -20
============================= test session starts ==============================
platform darwin -- Python 3.9.0, pytest-6.2.0, py-1.10.0, pluggy-0.13.1
Using --randomly-seed=239152410
rootdir: /Users/ajp/src/applandinc/appmap-python, configfile: pytest.ini
plugins: appmap-0.0.0, pyfakefs-4.3.2, datafiles-2.0, cov-2.11.1, randomly-3.5.0, xdist-2.2.0, forked-1.3.0, mock-3.5.1
collected 20 items
```